### PR TITLE
chore: improve codebase with some clippy suggestions

### PR DESCRIPTION
**Pull request by tcoratger**
## Description

I think this can't be properly caught by the CI before we bump Rust to something like 1.82 but these suggestions still apply and simplify the codebase

- [ ] Targeted PR against correct branch (master)
- [ ] Linked to GitHub issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Wrote unit tests
- [ ] Updated relevant documentation in the code
- [ ] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer

### CODE REVIEW

**Comment by tcoratger** [comment]
@Pratyush let me know if you want to merge this, even if the automatic detection for the future cannot be automated with clippy by the CI before we bump rust, I think this PR only brings simplifications :)

File: ec/src/hashing/curve_maps/wb.rs [hunk 1]
```diff
@@ -64,12 +65,16 @@ where
...
+///
+/// This method is used for curves in Weierstrass form defined by:
+///
+/// `y^2 = x^3 + a*x + b` where `b != 0`, but `a` can be zero,
+/// as seen in curves like BLS-381.
+///
+/// For more information, refer to [WB2019].
 ///
-/// - [\[WB2019\]] <http://dx.doi.org/10.46586/tches.v2019.i4.154-179>
+/// - [WB2019] <http://dx.doi.org/10.46586/tches.v2019.i4.154-179>
```
Comment (Pratyush) [inline ec/src/hashing/curve_maps/wb.rs:77]: We do want the square brackets around the reference

File: ec/src/hashing/tests.rs [hunk 2]
```diff
@@ -6,9 +6,9 @@ fn test_parity_of_prime_field_elements() {
     let a1 = Fq::from(0);
     let a2 = Fq::from(1);
     let a3 = Fq::from(10);
-    assert_eq!(parity(&a1), false);
-    assert_eq!(parity(&a2), true);
-    assert_eq!(parity(&a3), false);
+    assert!(!parity(&a1));
+    assert!(parity(&a2));
+    assert!(!parity(&a3));
```
Comment (Pratyush) [inline ec/src/hashing/tests.rs:11]: I think for these cases it's a bit easy to miss the `!`, so I prefer the old way

File: ec/src/hashing/tests.rs [hunk 3]
```diff
@@ -17,10 +17,10 @@ fn test_parity_of_quadratic_extension_elements() {
...
     let element_test3 = Fq2::new(Fq::from(10), Fq::from(5));
     let element_test4 = Fq2::new(Fq::from(5), Fq::from(10));
-    assert_eq!(parity(&element_test1), true, "parity is the oddness of first non-zero coefficient of element represented over the prime field" );
-    assert_eq!(parity(&element_test2), true);
-    assert_eq!(parity(&element_test3), false);
-    assert_eq!(parity(&element_test4), true);
+    assert!(parity(&element_test1), "parity is the oddness of first non-zero coefficient of element represented over the prime field" );
+    assert!(parity(&element_test2));
+    assert!(!parity(&element_test3));
+    assert!(parity(&element_test4));
```
Comment (Pratyush) [inline ec/src/hashing/tests.rs:23]: Ditto

File: ec/src/hashing/tests.rs [hunk 4]
```diff
@@ -37,9 +37,9 @@ fn test_parity_of_cubic_extension_elements() {
...
-    assert_eq!(parity(&element_test1), true, "parity is the oddness of first non-zero coefficient of element represented over the prime field");
-    assert_eq!(parity(&element_test2), true, "parity is the oddness of first non-zero coefficient of element represented over the prime field");
-    assert_eq!(parity(&element_test3), true);
-    assert_eq!(parity(&element_test4), true);
-    assert_eq!(parity(&element_test5), false);
+    assert!(parity(&element_test1), "parity is the oddness of first non-zero coefficient of element represented over the prime field");
+    assert!(parity(&element_test2), "parity is the oddness of first non-zero coefficient of element represented over the prime field");
+    assert!(parity(&element_test3));
+    assert!(parity(&element_test4));
+    assert!(!parity(&element_test5));
```
Comment (Pratyush) [inline ec/src/hashing/tests.rs:44]: Ditto

File: ec/src/models/bw6/mod.rs [hunk 5]
```diff
@@ -74,7 +74,9 @@ pub trait BW6Config: 'static + Eq + Sized {
 
     fn final_exponentiation(f: MillerLoopOutput<BW6<Self>>) -> Option<PairingOutput<BW6<Self>>> {
         let easy_part = BW6::<Self>::final_exponentiation_easy_part(f.0);
-        Some(Self::final_exponentiation_hard_part(&easy_part)).map(PairingOutput)
+        Some(PairingOutput(Self::final_exponentiation_hard_part(
+            &easy_part,
+        )))
```
Comment (Pratyush) [inline ec/src/models/bw6/mod.rs:79]: The old way seems to be cleaner, no?

File: ec/src/scalar_mul/glv.rs [hunk 6]
```diff
@@ -29,9 +31,8 @@ pub trait GLVConfig: Send + Sync + 'static + SWCurveConfig {
     ) -> ((bool, Self::ScalarField), (bool, Self::ScalarField)) {
         let scalar: BigInt = k.into_bigint().into().into();
 
-        let coeff_bigints: [BigInt; 4] = Self::SCALAR_DECOMP_COEFFS.map(|x| {
-            BigInt::from_biguint(x.0.then_some(Sign::Plus).unwrap_or(Sign::Minus), x.1.into())
-        });
+        let coeff_bigints: [BigInt; 4] = Self::SCALAR_DECOMP_COEFFS
+            .map(|x| BigInt::from_biguint(if x.0 { Sign::Plus } else { Sign::Minus }, x.1.into()));
```
Comment (Pratyush) [inline ec/src/scalar_mul/glv.rs:35]: Maybe we should just extract the sign into a separate variable? It's a bit noisy atm

File: ec/src/scalar_mul/mod.rs [hunk 7]
```diff
@@ -182,7 +182,7 @@ impl<T: ScalarMul> BatchMulPreprocessing<T> {
     ) -> Self {
         let window = Self::compute_window_size(num_scalars);
         let in_window = 1 << window;
-        let outerc = (max_scalar_size + window - 1) / window;
+        let outerc = max_scalar_size.div_ceil(window);
```
Comment (Pratyush) [inline ec/src/scalar_mul/mod.rs:185]: Nice!

File: ff/src/biginteger/tests.rs [hunk 8]
```diff
@@ -16,19 +17,19 @@ fn biginteger_arithmetic_test<B: BigInteger>(a: B, b: B, zero: B, max: B) {
     let mut a0_add = a;
     let carry = a0_add.add_with_carry(&zero);
     assert_eq!(a0_add, a);
-    assert_eq!(carry, false);
+    assert!(!carry);
```
Comment (Pratyush) [inline ff/src/biginteger/tests.rs:20]: Same here, the `!` is a bit easy to miss

File: poly/src/evaluations/multivariate/multilinear/dense.rs [hunk 9]
```diff
@@ -247,7 +247,7 @@ impl<F: Field> Add for DenseMultilinearExtension<F> {
     }
 }
 
-impl<'a, 'b, F: Field> Add<&'a DenseMultilinearExtension<F>> for &'b DenseMultilinearExtension<F> {
+impl<'a, F: Field> Add<&'a DenseMultilinearExtension<F>> for &DenseMultilinearExtension<F> {
```
Comment (Pratyush) [inline poly/src/evaluations/multivariate/multilinear/dense.rs:250]: Which lifetime does the `&DenseMultilinearExtension` get here? `'a` or an anonymous one?

File: poly/src/polynomial/univariate/sparse.rs [hunk 10]
```diff
@@ -139,21 +140,23 @@ impl<'a, 'b, F: Field> Add<&'a SparsePolynomial<F>> for &'b SparsePolynomial<F>
...
-                    result
-                        .coeffs
-                        .push((self_term_degree, self_term_coeff + other_term_coeff));
-                }
-                self_index += 1;
-                other_index += 1;
-            } else {
-                result.coeffs.push((other_term_degree, other_term_coeff));
-                other_index += 1;
+            match self_term_degree.cmp(&other_term_degree) {
```
Comment (Pratyush) [inline poly/src/polynomial/univariate/sparse.rs:143]: Nice refactor!

**Review by Pratyush — APPROVED** [review]

**Review by Pratyush — CHANGES_REQUESTED** [review]
Thanks for the PR! Just some small comments

File: ec/src/hashing/tests.rs (hunk 4 above)
Comment (tcoratger) [inline ec/src/hashing/tests.rs:44]:
```suggestion
    assert_eq!(parity(&element_test1), true, "parity is the oddness of first non-zero coefficient of element represented over the prime field");
    assert_eq!(parity(&element_test2), true, "parity is the oddness of first non-zero coefficient of element represented over the prime field");
    assert_eq!(parity(&element_test3), true);
    assert_eq!(parity(&element_test4), true);
    assert_eq!(parity(&element_test5), false);
```

Comment (tcoratger) [reply ec/src/hashing/tests.rs]: fixed

File: ec/src/hashing/tests.rs (hunk 2 above)
Comment (tcoratger) [reply ec/src/hashing/tests.rs]: fixed

File: ec/src/hashing/curve_maps/wb.rs (hunk 1 above)
Comment (tcoratger) [reply ec/src/hashing/curve_maps/wb.rs]: fixed

File: poly/src/evaluations/multivariate/multilinear/dense.rs (hunk 9 above)
Comment (tcoratger) [reply poly/src/evaluations/multivariate/multilinear/dense.rs:250]: It's an anonymous lifetime directly inferred by the compiler :)

File: ec/src/scalar_mul/glv.rs (hunk 6 above)
Comment (tcoratger) [reply ec/src/scalar_mul/glv.rs]: fixed

**Comment by tcoratger** [comment]
> Thanks for the PR! Just some small comments

@Pratyush Should be all fixed, let me know if something is missing :)

File: ff/src/biginteger/tests.rs [hunk 11]
```diff
@@ -113,9 +114,9 @@ fn biginteger_shr<B: BigInteger>() {
     // Test null bits
     let a = B::rand(&mut rng);
     let b = a >> 3;
-    assert_eq!(b.get_bit(B::NUM_LIMBS * 64 - 1), false);
-    assert_eq!(b.get_bit(B::NUM_LIMBS * 64 - 2), false);
-    assert_eq!(b.get_bit(B::NUM_LIMBS * 64 - 3), false);
+    assert!(!b.get_bit(B::NUM_LIMBS * 64 - 1));
+    assert!(!b.get_bit(B::NUM_LIMBS * 64 - 2));
+    assert!(!b.get_bit(B::NUM_LIMBS * 64 - 3));
```
Comment (Pratyush) [inline ff/src/biginteger/tests.rs:119]:
```suggestion
    assert_eq!(b.get_bit(B::NUM_LIMBS * 64 - 1), false);
    assert_eq!(b.get_bit(B::NUM_LIMBS * 64 - 2), false);
    assert_eq!(b.get_bit(B::NUM_LIMBS * 64 - 3), false);
```

File: ff/src/biginteger/tests.rs [hunk 12]
```diff
@@ -134,9 +135,9 @@ fn biginteger_shl<B: BigInteger>() {
     // Test null bits
     let a = B::rand(&mut rng);
     let b = a << 3;
-    assert_eq!(b.get_bit(0), false);
-    assert_eq!(b.get_bit(1), false);
-    assert_eq!(b.get_bit(2), false);
+    assert!(!b.get_bit(0));
+    assert!(!b.get_bit(1));
+    assert!(!b.get_bit(2));
```
Comment (Pratyush) [inline ff/src/biginteger/tests.rs:140]:
```suggestion
    assert_eq!(b.get_bit(0), false);
    assert_eq!(b.get_bit(1), false);
    assert_eq!(b.get_bit(2), false);
```

File: ff/src/biginteger/tests.rs [hunk 13]
```diff
@@ -22,13 +23,13 @@ fn biginteger_arithmetic_test<B: BigInteger>(a: B, b: B, zero: B, max: B) {
     let mut a0_sub = a;
     let borrow = a0_sub.sub_with_borrow(&zero);
     assert_eq!(a0_sub, a);
-    assert_eq!(borrow, false);
+    assert!(!borrow);
```
Comment (Pratyush) [inline ff/src/biginteger/tests.rs:26]:
```suggestion
    assert_eq!(borrow, false);
```

File: ff/src/biginteger/tests.rs [hunk 14]
```diff
@@ -22,13 +23,13 @@ fn biginteger_arithmetic_test<B: BigInteger>(a: B, b: B, zero: B, max: B) {
...
 
     // a - a = 0
     let mut aa_sub = a;
     let borrow = aa_sub.sub_with_borrow(&a);
     assert_eq!(aa_sub, zero);
-    assert_eq!(borrow, false);
+    assert!(!borrow);
```
Comment (Pratyush) [inline ff/src/biginteger/tests.rs:32]:
```suggestion
    assert_eq!(borrow, false);
```

**Review by Pratyush — APPROVED** [review]

**Comment by tcoratger** [comment]
@Pratyush ci failure seems unrelated no?

**Pull request merged**